packages/api/app/Models/Permission.js

```diff
@@ -80,6 +80,19 @@ class Permission extends Model {
 				return false;
 			}
 		}
+		if (matchesPermission([permissions.LIST_TECHNOLOGY_COMMENTS], matchedPermission)) {
+			const technologyInst = await Technology.findOrFail(techonologyResourceId);
+			const isResponsible = await technologyInst.checkResponsible(user);
+			if (isResponsible) return true;
+			const reviewer = await technologyInst.getReviewer();
+			if (reviewer) {
+				const reviewerUser = await reviewer.user().first();
```

Comment:
```suggestion
				const reviewerUser = await reviewer.user().query().select('id').first();
```

Reply:
Rapaz deu um erro aqui kk
`reviewer.user(...).query is not a function`

Reply:
A lógica aqui é a seguinte, da tecnologia eu pego o reviewer, do reviewer eu pego o user, ai eu comparo se o user que está enviando a requisição é reviewer da tecnologia.

Reply:
Uma otimização do processo seria substituir toda parte que checa o reviewer por essa query:
```javascript
const isReviewer = await Reviewer.query()
				.whereHas('technologies', (builder) => {
					builder.where('id', technologyInst.id);
				})
				.where({
					user_id: user.id,
				})
				.first();
```

Reply:
O erro é só questão de sintaxe mesmo 😅. A sugestão se resume a **trazer do banco somente os dados que usaremos**, no caso, somente o id. Se conseguir aplicar isso com a sintaxe do query builder do Adonis seria ótimo pois atualmente trazemos todas as colunas da tabela pra só usarmos o id.

Reply:
Se conseguir diminuir pra menos queries com essa sugestão acima, melhor ainda! 🚀

Reply:
Melhorou porque eu substitui bastante código por essa query.